AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/rclass/RInnerClass.java

```diff
@@ -122,9 +124,30 @@ public static JFieldRef extractIdStaticRef(AndroidAnnotationsEnvironment environ
 			int fieldSuffix = layoutFieldQualifiedName.lastIndexOf('.');
 			String fieldName = layoutFieldQualifiedName.substring(fieldSuffix + 1);
 			String rInnerClassName = layoutFieldQualifiedName.substring(0, fieldSuffix);
+			int innerClassSuffix = rInnerClassName.lastIndexOf('.');
 
-			AbstractJClass refClass = environment.getJClass(rInnerClassName);
-			return refClass.staticRef(fieldName);
+			String rClassQualifiedName = rInnerClassName.substring(0, innerClassSuffix);
+			String innerClassSimpleName = rInnerClassName.substring(innerClassSuffix + 1);
+
+			JDirectClass rClass = (JDirectClass) environment.getJClass(rClassQualifiedName);
+
+			AbstractJClass innerClass = null;
+
+			for (JDirectClass clazz : rClass.classes()) {
+				if (clazz.name().equals(innerClassSimpleName)) {
+					innerClass = clazz;
+					break;
+				}
+			}
+			if (innerClass == null) {
+				try {
+					innerClass = rClass._class(innerClassSimpleName);
+				} catch (JClassAlreadyExistsException e) {
+					// never happens, since we already checked the inner class does not exist
```

Comment:
If it never happens, I think we should throw a `RuntimeException`. So that, it will crash and users will report it.